serviceregistration/kubernetes/testing/README.md

```diff
@@ -0,0 +1,54 @@
+# How to Test Manually
+
+- `$ minikube start`
+- In the Vault folder, `$ make dev`
+- Create a file called `vault-test.yaml` with the following contents:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: vault
+spec:
+  containers:
+    - name: nginx
+      image: nginx
+      command: [ "sh", "-c"]
+      args:
+      - while true; do
+          echo -en '\n';
+          printenv VAULT_POD_NAME VAULT_NAMESPACE;
+          sleep 10;
+        done;
+      env:
+        - name: VAULT_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: VAULT_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+  restartPolicy: Never
+```
+
+- Create the pod: `$ kubectl apply -f vault-test.yaml`
+- View the full initial state of the pod: `$ kubectl get pod vault -o=yaml > initialstate.txt`
+- Drop the Vault binary into the pod: `$ kubectl cp $(which vault) /vault:/`
+- Drop to the shell within the pod: `$ kubectl exec -it vault -- /bin/bash`
+- Install a text editor: `$ apt-get update`, `$ apt-get install nano`
+- Write a test Vault config to `vault.config` like:
+
+```
+storage "inmem" {}
+service_registration "kubernetes" {}
+disable_mlock = true
+ui = true
+api_addr = "http://127.0.0.1:8200"
+log_level = "debug"
+```
+
+- Run Vault: `$ ./vault server -config=vault.config -dev -dev-root-token-id=root`
```

Comment:
I had to set these vars too FWIW:
```suggestion
- Set env vars: `$ export VAULT_K8S_NAMESPACE=$VAULT_NAMESPACE VAULT_K8S_POD_NAME=$VAULT_POD_NAME`
- Run Vault: `$ ./vault server -config=vault.config -dev -dev-root-token-id=root`
```

Reply:
Thanks for mentioning that! That actually means I didn't properly update the yaml for making the pod above. Fixed in https://github.com/hashicorp/vault/pull/8249/commits/e2a0a7b07bb746bf7fb49b79f3aae3b354067d71